### ISSUE TITLE
Infinite Recursion browser tab crashes

### ISSUE THREAD

**Issue by habeelmazhar** [ali-saleem/anchorme.js#115]
Any URL that ends with some characters like `@` or `$` crashes.
[Codepen link with an example](https://codepen.io/habeelmazhar/pen/jOmKRWd)

Obviously codepen URL does not crash but you can see that the script execution gets stuck.

**Comment by zahassanyum** [comment]
Is this related to: https://github.com/alexcorvi/anchorme.js/issues/82?

@habeelmazhar

**Comment by habeelmazhar** [comment]
@zahassanyum Yeah almost the same, but it's different in a way that encodeURI does not fix this issue

**Comment by lionel-rowe** [comment]
It's a [Catastrophic Backtracking](https://www.regular-expressions.info/catastrophic.html) issue. It's not technically "infinite", it's just that execution time grows exponentially with the length of the path, such that `https://example.com/abc@` validates in reasonable time but `https://example.com/abcdefghijklmnopqrstuvwxyz123456789@` crashes.
